mailbox/store/src/main/java/org/apache/james/mailbox/store/SessionProviderImpl.java

```diff
@@ -57,38 +55,38 @@ public MailboxSession createSystemSession(Username userName) {
     }
 
     @Override
-    public AuthorizationStep authenticate(Username thisUserId, String passwd) {
+    public AuthorizationStep authenticate(Username loginUsername, String passwd) {
         return new AuthorizationStep() {
             @Override
             public MailboxSession as(Username otherUserId) throws MailboxException {
-                if (!isValidLogin(thisUserId, passwd)) {
+                Optional<Username> loggedInUser = isValidLogin(loginUsername, passwd);
```

Comment:
lambda beauty 
```suggestion
return isValidLogin(loginUsername, passwd)
                    .map(loggedInUser -> authenticate(loggedInUser).as(otherUserId))
                    .orElseThrow(BadCredentialsException::new);
```

Reply:
This method signature explicitly throws checked `MailboxException` and the same for `authenticate(loggedInUser).as(otherUserId)`. We can not throw a checked exception in lambda. If we use Throwing.function, we would turn the checked exception into an unchecked exception, and I am not sure we want that here.
Your opinion @vttranlina ?

Reply:
checked exception, then ignore previous my comment